chains/substrate/executor/message-handler.go

```diff
@@ -68,14 +73,40 @@ func FungibleTransferMessageHandler(m *message.Message) (*proposal.Proposal, err
 	if !ok {
 		return nil, errors.New("wrong payload amount format")
 	}
-	recipient, ok := m.Payload[1].([]byte)
+	reciever, ok := m.Payload[1].([]byte)
 	if !ok {
 		return nil, errors.New("wrong payload recipient format")
 	}
 	var data []byte
 	data = append(data, common.LeftPadBytes(amount, 32)...) // amount (uint256)
-	recipientLen := big.NewInt(int64(len(recipient))).Bytes()
-	data = append(data, common.LeftPadBytes(recipientLen, 32)...) // length of recipient (uint256)
-	data = append(data, recipient...)                             // recipient ([]byte)
+	acc := *(*[]types.U8)(unsafe.Pointer(&reciever))
```

Comment:
Move this logic somewhere else into some function.
Looks ugly and polutes the code.